routes/api.js

```diff
@@ -199,8 +216,10 @@ var killKernel = function(kernelId) {
         method: 'DELETE',
         headers: headers
     }, function(err, response, body) {
-        debug('PROXY: kill kernel response: ' +
-            response.statusCode + ' ' + response.statusMessage);
+        if (response) {
```

Comment:
So far I think that's only happened because the test was originally a spy. I've updated it to be a stub instead, and have a "response" return value.